openssh/README.md

```diff
@@ -0,0 +1,106 @@
+3. `docker run --rm --name oqs-openssh-client -dit oqs-openssh-img` will start a docker container with the same properties as `oqs-openssh-server` except the port 2222 is not published.
+4. You can hop on either of those two containers as a non-root user (oqs) to use the built-in OQS-OpenSSH binaries or do other shenanigans by typing
+`docker exec -ti -u oqs oqs-openssh-server /bin/sh`
+Of course adjust the container's name accordingly if hopping onto the client (==> `oqs-openssh-client`).
+
+It is possible that the `docker build [...]` command fails with something like `Got permission denied while trying to connect to the Docker daemon socket [...]`. That is because the active user is not a member of the `docker` group yet. To fix this run:
+```bash
+usermod -aG docker <user>
+newgrp docker
+```
+The first command adds user `<user>` to the group `docker`, and the second simulates a logout/login, so you don't have to do a re-login.
+
+### Connect from container to container via OQS-SSH
+
+1. To connect to another docker container, we first need to create a docker network by typing `docker network create oqs-openssh-net`.
+
+1. Then connect the containers to this network by typing `docker network connect oqs-openssh-net <name-of-container>` where `<name-of-container>` is once the server's and once the client's name. Obviously this does not work if the containers are not yet running, in this case refer to the [Quick Start](README.md#quick-start) section.
+
+1. Then connect to the server by typing `docker exec -ti oqs-openssh-client ssh oqs@oqs-openssh-server` and authenticating the user `oqs` with its default password `oqs.pw`.
+
+As server and client are based on the same image, connecting from the server to the client's ssh daemon is possible as well. For that use the same command and exchange `server` and `client`.
+
+## More details
+
+The Dockerfile 
+- obtains all source code required for building the quantum-safe cryptography (QSC) algorithms and the [QSC-enabled version of OpenSSH (7.9-2020-08_p1)](https://github.com/open-quantum-safe/openssh/releases/tag/OQS-OpenSSH-snapshot-2020-08)
+- builds all libraries and applications
+- by default starts the openssh daemon\*
+- creates a second user `oqs` with the default password `oqs.pw`
+- by default creates host keys based on the enabled host key algorithms in `sshd_config`\*
+- by default creates identity keys based on the config file `ssh_config`\*
+
+\*those steps are executed when the image is startet.
+
+**Note for the interested**: The build process is two-stage with the final image only retaining all executables, libraries and include-files to utilize OQS-enabled openssh.
+
+The re-generation of the host and identity keys happens via the script [key-regen.sh](key-regen.sh) that is called with a `CMD` command at the end of the [Dockerfile](Dockerfile). The script checks if the required key already exist and gerenates it if necessary.
```

Comment:
"The script checks if the required key already exist and gerenates it if necessary."

What happens if it already exists? The name of the script makes it sound like it will be overwritten. If it's not the desired outcome, maybe the script should be named `key-gen.sh`. If it is as expected, I'd remove the "if already exist" mention in the text.

Reply:
True that. No it does not regenerate keys, or not anymore. Before I started using the `install-nokeys` build target it was regenerating keys because a Docker image with hard coded keys ain't really best practice from a security standpoint.

Said that I will rename it to reflect what it is actually doing.